integration-tests/bats/checkout.bats

```diff
@@ -285,3 +287,68 @@ SQL
   [[ "$commitmeta" =~ "$shaparent1" ]] || false
   [[ "$commitmeta" =~ "$shaparent2" ]] || false
 }
+
+
+@test "checkout: block checkout when current and target branches have working set changes" {
+  dolt sql -q "create table users (id int primary key, name varchar(32));"
+  dolt add .
+  dolt commit -m "original users table"
+  dolt branch -c main feature
+
+  # make changes on main
+  dolt sql -q 'insert into users (id, name) values (1, "main-change");'
+  # make sure changes are present
+  run dolt sql -q "select count(*) from users"
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "1" ]] || false
+
+  # checkout feature
+  dolt checkout feature
+  # make sure changes are pulled in from main
+  run dolt sql -q "select count(*) from users"
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "1" ]] || false
+
+  # make changes on feature
+  dolt sql -q 'insert into users (id, name) values (2, "feature-change");'
+  # make sure new changes are present
+  run dolt sql -q "select count(*) from users"
+  [ "$status" -eq 0 ]
+  [[ "$output" =~ "2" ]] || false
+
+  # try to checkout main, but fail due to working set changes
+  run dolt checkout main
```

Comment:
This isn't the best experience for the command line use case. It will happen pretty often to people who switch branches, and it's confusing. You can move forward (`dolt checkout -b branch1`,  `dolt checkout -b branch2`), but then as soon as you move backwards it's an error. Not great.

I think we should pursue the approach we talked about briefly, where checking out a branch on the CLI actually implicitly reverts any uncommitted changes on the current branch before doing the checkout. Then your have working set changes on exactly one branch at a time.

Reply:
That seems reasonable.

So if you are on `main` and have unstaged changes, and are switching to `feature` that also has unstage changes, the behavior should be:

- `dolt checkout feature` (no `-f` flag): revert unstaged `main` changes, switch to `feature` and see just `feature` changes
- `dolt checkout -f feature` (`-f` flag present): switch to `feature` and bring in `feature` changes, which are now combined with `main` changes

The behavior would also mean that this operation is a no-op:
```
dolt checkout feature
dolt checkout main
```
Because both main and feature keep their working sets despite the checkouts.

Correct?